src/objectTrie.js

```diff
@@ -0,0 +1,23 @@
+import {weakMemoizeArray} from "./weakMemoize";
+
+export const EDGE = 'EDGE';
+
+const buildObjTrie = (lines) => {
+  const root = {};
+  for (let i = 0; i < lines.length; ++i) {
+    const path = lines[i].split('.');
+    let node = root;
+    const lastIndex = path.length - 1;
+    for (let j = 1; j < lastIndex; ++j) {
+      const item = path[j];
+      if (!node[item] || node[item] === EDGE) {
+        node[item] = {};
+      }
+      node = node[item];
+    }
+    node[path[lastIndex]] = EDGE;
+  }
+  return root; // FIXME
```

Comment:
This FIXME comment was there because I thought you don't want to use `weakMemoizeArray` but something like `weakMemoizeObject` which can eliminate `if (old.length === arg.length) {` check.
If it's not a big issue, please just remove the FIXME comment.